synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -701,15 +701,17 @@ def decode_and_calc_og(
     try:
         parser = etree.HTMLParser(recover=True, encoding=request_encoding)
         tree = etree.fromstring(body, parser)
-        og = _calc_og(tree, media_uri)
     except UnicodeDecodeError:
         # blindly try decoding the body as utf-8, which seems to fix
         # the charset mismatches on https://google.com
         parser = etree.HTMLParser(recover=True, encoding=request_encoding)
         tree = etree.fromstring(body.decode("utf-8", "ignore"), parser)
-        og = _calc_og(tree, media_uri)
+    except LookupError:
+        # blindly try decoding the body as utf-8.
+        parser = etree.HTMLParser(recover=True, encoding="utf-8")
+        tree = etree.fromstring(body, parser)
```

Comment:
Do we want to wrap this in a `try/except` so we can handle the error if this fails? Or does that happen higher up?

Reply:
There isn't additional error handling. I'm not really sure what this will throw unfortunately, and the lxml docs are a bit lacking. I'll take a look though and see if there's something reasonable we can do here.

Reply:
I think when we're calling into a library its probably fine to just have a generic `except Exception:`?